pkg/controllerutils/miscellaneous.go

```diff
@@ -124,3 +128,25 @@ func ReconcileOncePer24hDuration(clock clock.Clock, objectMeta metav1.ObjectMeta
 
 	return 0
 }
+
+// GetMainReconciliationContext returns context with timeout for main client. Timeout should be equal to timeout passed in the argument but
+// not more than the DefaultReconciliationTimeout.
```

Comment:
"Timeout should be equal..." is a bit confusing, IMO.

```suggestion
// GetMainReconciliationContext returns a context with timeout for the controller's main client. The resulting context has a timeout equal to the timeout passed in the argument but
// not more than DefaultReconciliationTimeout.
```